packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx

```diff
@@ -51,6 +57,147 @@ type CWRequirementsRadioButtonProps = {
+  address: string,
+  network_url,
+): Promise<number | null> {
+  try {
+    const provider = new ethers.providers.JsonRpcProvider(network_url);
+
+    const contractERC20 = new ethers.Contract(address, Abis.ERC20, provider);
+
+    const decimals = await contractERC20.decimals();
+    return decimals;
+  } catch (error) {
+    console.error(error);
+    return null;
+  }
+}
+
+async function isEVMAddressContract(
+  address: string,
+  network_url: string,
+): Promise<boolean> {
+  const provider = new ethers.providers.JsonRpcProvider(network_url);
+  try {
+    const code = await provider.getCode(address);
+    return code !== '0x';
+  } catch (error) {
+    console.error('Error checking address:', error);
+    return false;
+  }
+}
+
+async function isCosmosAddressContract(
+  address: string,
+  network_url: string,
+): Promise<boolean> {
+  try {
+    const client = new HttpClient(network_url);
+    const tmClient = await Tendermint34Client.create(client);
+    const api = QueryClient.withExtensions(tmClient, setupWasmExtension);
+
+    const contractInfo = await api.wasm.getContractInfo(address);
+    if (contractInfo) {
+      return true;
+    } else {
+      return false;
+    }
+  } catch (error) {
+    return false;
+  }
+}
+
+const getEVMContractType = async (
+  contractAddress,
+  network_url,
+): Promise<string | null> => {
+  const provider = new ethers.providers.JsonRpcProvider(network_url);
+
+  const contractERC20 = new ethers.Contract(
+    contractAddress,
+    Abis.ERC20,
+    provider,
+  );
+  const contractERC721 = new ethers.Contract(
+    contractAddress,
+    Abis.ERC721,
+    provider,
+  );
+  const contractERC1155 = new ethers.Contract(
+    contractAddress,
+    Abis.ERC1155,
+    provider,
+  );
+
+  try {
+    // Check for ERC20
+    await contractERC20.decimals();
+    return 'erc20';
+  } catch (e) {
+    /* Not ERC20 */
+  }
+
+  try {
+    // Check for ERC721
+    await contractERC721.balanceOf(ethers.constants.AddressZero);
+    return 'erc721';
+  } catch (e) {
+    /* Not ERC721 */
+  }
+
+  try {
+    // Check for ERC1155
+    await contractERC1155.isApprovedForAll(
+      ethers.constants.AddressZero,
+      ethers.constants.AddressZero,
+    );
+    return 'erc1155';
+  } catch (e) {
+    /* Not ERC1155 */
+  }
+
+  console.log(
+    'Contract type is unknown or does not implement standard interfaces.',
+  );
+  return null;
+};
+
+async function getCosmosContractType(
+  address: string,
+  network_url: string,
+): Promise<string | null> {
+  try {
+    const client = new HttpClient(network_url);
+    const tmClient = await Tendermint34Client.create(client);
+    const api = QueryClient.withExtensions(tmClient, setupWasmExtension);
+
+    const contractInfo = await api.wasm.getContractInfo(address);
+    if (contractInfo && contractInfo.contractInfo.codeId.equals(49)) {
+      return 'cw721';
+    } else {
+      return null;
+    }
+  } catch (error) {
+    return null;
+  }
+}
+
```

Comment:
1. please use `const x = () => {}` instead of function => this is our convention and we want to keep the code style in the same way all over the front end places
2. please move all of these functions to `Groups/common/helpers/index.ts`. `GroupForm.tsx` should be as concise as possible.